Formula/subversion-javahl.rb

```diff
@@ -0,0 +1,182 @@
+        s.gsub! "'SERF_MINOR_VERSION ([0-9]+).*'",
+        "b'SERF_MINOR_VERSION ([0-9]+).*'"
+        s.gsub! "'SERF_PATCH_VERSION ([0-9]+)'",
+        "b'SERF_PATCH_VERSION ([0-9]+)'"
+        s.gsub! "variables=opts,",
+        "variables=opts, RPATHPREFIX = '-Wl,-rpath,',"
+      end
+
+      # scons ignores our compiler and flags unless explicitly passed
+      krb5 = if OS.mac?
+        "/usr"
+      else
+        Formula["krb5"].opt_prefix
+      end
+
+      args = %W[
+        PREFIX=#{serf_prefix} GSSAPI=#{krb5} CC=#{ENV.cc}
+        CFLAGS=#{ENV.cflags} LINKFLAGS=#{ENV.ldflags}
+        OPENSSL=#{Formula["openssl@1.1"].opt_prefix}
+        APR=#{Formula["apr"].opt_prefix}
+        APU=#{Formula["apr-util"].opt_prefix}
+      ]
+
+      args << "ZLIB=#{Formula["zlib"].opt_prefix}" if OS.linux?
+
+      scons = Formula["scons"].opt_bin/"scons"
+      system scons, *args
+      system scons, "install"
+    end
+
+    # Use existing system zlib and sqlite
+    zlib = if OS.mac?
+      "#{MacOS.sdk_path_if_needed}/usr"
+    else
+      Formula["zlib"].opt_prefix
+    end
+
+    sqlite = if OS.mac?
+      "#{MacOS.sdk_path_if_needed}/usr"
+    else
+      Formula["sqlite"].opt_prefix
+    end
+
+    # Use dep-provided other libraries
+    # Don't mess with Apache modules (since we're not sudo)
+    if OS.linux?
+      # svn can't find libserf-1.so.1 at runtime without this
+      ENV.append "LDFLAGS", "-Wl,-rpath=#{serf_prefix}/lib"
+    end
+
+    openjdk = deps.map(&:to_formula).find { |f| f.name.match? "^openjdk" }
+
+    args = %W[
+      --prefix=#{prefix}
+      --disable-debug
+      --enable-optimize
+      --disable-mod-activation
+      --disable-plaintext-password-storage
+      --with-apr-util=#{Formula["apr-util"].opt_prefix}
+      --with-apr=#{Formula["apr"].opt_prefix}
+      --with-apxs=no
+      --with-py3c=#{py3c_prefix}
+      --with-serf=#{serf_prefix}
+      --with-sqlite=#{sqlite}
+      --with-zlib=#{zlib}
+      --without-apache-libexecdir
+      --without-berkeley-db
+      --without-gpg-agent
+      --without-jikes
+      --with-jdk=#{Formula["openjdk"].opt_prefix}
+      --enable-javahl
+    ]
+
+    # preserve compatibility with macOS 12.0–12.2
+    args.unshift "--enable-sqlite-compatibility-version=3.36.0" if MacOS.version == :monterey
+
+    inreplace "Makefile.in",
+              "toolsdir = @bindir@/svn-tools",
+              "toolsdir = @libexecdir@/svn-tools"
+
+    # regenerate configure file as we patched `build/ac-macros/swig.m4`
+    system "./autogen.sh" if build.head?
+    system "./configure", *args
+    system "make"
+
+    ENV.deparallelize
+    system "make", "javahl"
+    system "make", "install-javahl"
+  end
+
+  def caveats
+    <<~EOS
+      You may need to link the Java bindings into the Java Extensions folder:
+        sudo mkdir -p /Library/Java/Extensions
+        sudo ln -s #{HOMEBREW_PREFIX}/lib/libsvnjavahl-1.dylib /Library/Java/Extensions/libsvnjavahl-1.dylib
```

Comment:
```suggestion
        sudo ln -s #{HOMEBREW_PREFIX}/lib/libsvnjavahl-#{version.major}.dylib /Library/Java/Extensions/libsvnjavahl-#{version.major}.dylib
```

Reply:
I think that's being very optimistic about subversion development. Also, I have no idea if they ever make a version 2, whether the javahl libraries would be affected as well?

Reply:
😄 I just think it looked nicer